### PR TITLE
fix(app-shell): Update Electron and add macOS app notarization

### LINKED ISSUE #2567: Install the App for all users
Is it possible to install the App on Windows for all users of the computer?

### LINKED ISSUE #3997: App release: Update macOS release process for notarization
## overview

With the next release of macOS (10.15 Catalina, Q3 2019), [notarization](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution) is going to be required for all apps.

## acceptance criteria

- [ ] Mac app releases are submitted to Apple for notarization and notary ticket is "stapled" to release

## implementation details

See the following links:

- https://github.com/electron/electron-notarize
- https://kilianvalkhof.com/2019/electron/notarizing-your-electron-application/
- https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_cs_allow-unsigned-executable-memory 

This could be a good time to implement our much-discussed alignment of the app release process with the web release process, where:

1. CI builds to sandbox bucket
2. sandbox artifacts are sanity checked
3. sandbox artifacts are "promoted" to staging / production via an S3 copy

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR updates a lot of our underlying Electron / app packaging tooling to ensure we are ready for new security requirements in the upcoming version of macOS (10.15 Catalina).

This Electron update will bring reliability and security improvements to all our users, but out of necessity, is going to introduce a little bit of user pain for our Linux and Windows users out of the gate in exchange for a better long term experience.

Closes #3997, closes #2567 

### the good

- When macOS 10.15 is launched in the coming weeks, our users won't get scary security alerts
- Our macOS distributable is switching from ZIP to DMG
    - Now, when users double-click their Mac download, they'll get instructions to drag the app to their `/Applications` folder
- Windows users can now choose to install for all users (if they have admin permission, see #2567)
- The app will be on Node v12 and Chromium v76, which is always nice for us (e.g. built-in `flatMap`!)
- Copy / paste works in the devtools now!

### the bad

- Support for macOS 10.9 Mavericks has been dropped
- ~We need to recommend to Windows users that they **uninstall 3.11.x** before installing this update to avoid weird problems caused by installation location bugs fixed in electron-builder~
    - See electron-userland/electron-builder#4092
    - As a workaround, we're going to keep Windows builds locked to electron-builder@21.1.2 until this bug is resolved so that our users don't have to deal with it
- Our macOS builds are going to take about 5 to 10 minutes longer because of the notarization process

### wash

- The installation experience of the Linux app is changing
    - Bad news: we need to direct them to install [AppImageLauncher](https://github.com/TheAssassin/AppImageLauncher) once, before downloading our app
    - Good news: once they've got AppImageLauncher installed, all they have to do to install the app is to double-click the download
    - In other words, **no more messing with file permissions before launching**; AppImageLauncher will just handle it

## changelog

- Update electron and electron-builder dependencies
- Update electron-builder configuration to support macOS notarization

## review requests

This PR is going to require stress testing on all of our OS's. Given that we don't have a choice on this update (because of macOS security), I think as long as we do various smoke tests with this PR, we should rely on our release QA to really make sure this is solid. Scariest areas that the Electron side of things deals with that we'll need to keep an eye on:

- App self-update
- App config
- Robot discovery
- Buildroot updates

### CODE REVIEW

**Comment by Kadee80** [comment]
Making a note about this here so it doesn't get forgotten:

OT App page on website currently lists supported OS as `macOS 10.9 or later`
Once this is merged and released we will need to update that to `macOS 10.10 or later`

@mcous please confirm i'm correct here with the new version support

**Comment by mcous** [comment]
@Kadee80 that is correct. Filed Opentrons/opentrons-dot-com#230 to track
